src/bigunit.ts

```diff
@@ -399,23 +406,28 @@ export class BigUnit {
     // If the length is less than the precision, pad with zeros
     if (length < this.precision) {
       const padding = "0".repeat(this.precision - length);
-      return `0.${padding}${valueString}`;
+      return isNegative 
+        ? `-0.${padding}${valueString}` 
+        : `0.${padding}${valueString}`;
     }
 
     // If the length is greater than the precision, slice the string
     const decimalIndex = length - this.precision;
     const integerPart = valueString.slice(0, decimalIndex);
     const fractionalPart = valueString.slice(decimalIndex);
-    return `${integerPart}.${fractionalPart}`;
+
+    // Construct the final string
+    const result = `${integerPart === "" ? "0" : integerPart}.${fractionalPart}`;
+    return isNegative ? `-${result}` : result;
   }
 
   public toBigInt(): bigint {
     return this.value;
   }
 
   /**
-   * @description Format the value as a string with the given precision
-   * @param precision
+   * @description Format the value as a string with the given precision (Note: this uses standard rounding)
+   * @param precision - the number of digits to appear after the decimal point
```

Comment:
I thought you were going to rewrite this one?

Reply:
I was but Mitch mentioned there was a ticket already specifically about this and that we needed to have a further discussion about how rounding works. So I have left it as is for now